src/hotspot/share/gc/parallel/psParallelCompact.hpp

```diff
@@ -520,16 +520,16 @@ inline size_t
 ParallelCompactData::region_offset(const HeapWord* addr) const
 {
   assert(addr >= _heap_start, "bad addr");
-  // would mistakenly return 0 for _region_end
+  // would mistakenly return 0 for _heap_end
```

Comment:
(pre-existing): I think this comment should be removed as it is really confusing in the current context because it refers to deleted code.

It seems that previously the next assert used `<=` instead of `<` which could cause the mentioned problem. However without context (like "Using <= would mistakenly...") this comment makes no sense. So I recommend removing this comment because even such a fixed comment would not help a lot imo.